src/Auth0.Core/Http/RateLimit.cs

```diff
@@ -21,5 +23,34 @@ public class RateLimit
         /// The date and time offset at which the current rate limit window is reset.
         /// </summary>
         public DateTimeOffset Reset { get; internal set; }
+
+        public static RateLimit Parse(HttpResponseHeaders headers)
```

Comment:
Is this available on all requests? Not on a rate limit hit only?

Reply:
It is, but we will no longer expose it unless it's an error.  Realistically there's not much you can do with it as it's a global rate. You can't co-ordinate between various clients to stay under.  If it were a per-instance rate limit it might make sense for simple single-threaded apps to expose it.

Reply:
> It is, but we will no longer expose it unless it's an error.

If this is only exposed in an error then the amount remaining and the limit itself is irrelevant. 

> You can't co-ordinate between various clients to stay under.

If you see the remaining/reset before, you could pause, if that made sense for the app.

Reply:
But only if it was the only instance of the app running anywhere.  You can't tell what the other clients are going to do and it's a per-tenant rate.